src/vs-bicep/Bicep.VSLanguageServerClient/ContentType/BicepConfigContentTypeDefinitionImpl.cs

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.ComponentModel.Composition;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Bicep.VSLanguageServerClient.ContentType
+{
+    public class BicepConfigContentTypeDefinitionImpl
+    {
+        /// <summary>
+        /// Exports the bicep config content type.
+        /// </summary>
+        [Export(typeof(ContentTypeDefinition))]
+        [Name(BicepConfigContentTypeDefinition.ContentType)]
+        [BaseDefinition(ExternalContentTypeDefinition.CodeRemoteContentTypeName)] // required for ILanguageClient support
```

Comment:
>ILanguageClient [](http://example.com/codeflow?start=98&length=15)

Will VS use the JSON language server with bicepconfig.json? (Do content types have multiple inheritance?)

Reply:
Looks like this below handles this:
> if (fileName.Equals("bicepconfig.json", StringComparison.Ordinal))

Reply:
Thanks Stephen!

@majastrz , I added BaseDefinition here - https://github.com/Azure/bicep/blob/9fec68204f2c6fae147424d4b7552f2ec16bfea9/src/vs-bicep/Bicep.VSLanguageServerClient/ContentType/BicepConfigContentTypeDefinitionImpl.cs#L17. This makes sure we get all the core VS editor features like validation, formatting, outlining, colorization etc. Completion is not working and I logged https://github.com/Azure/bicep/issues/7766 to track the issue.

This class - https://github.com/Azure/bicep/blob/9fec68204f2c6fae147424d4b7552f2ec16bfea9/src/vs-bicep/Bicep.VSLanguageServerClient/ContentType/BicepConfigContentTypeProvider.cs#L32 makes sure we only handle file changes for file with name "bicepconfig.json" (case insensitive).

I added a demo .gif. Let me know if you have any questions.

Thanks!